dig_test.go

```diff
@@ -3566,3 +3567,91 @@ func TestEndToEndSuccessWithAliases(t *testing.T) {
 	})
 
 }
+
+func TestConcurrency(t *testing.T) {
+	// Ensures providers will run at the same time
+	t.Run("TestMaxConcurrency", func(t *testing.T) {
+		t.Parallel()
+
+		type (
+			A int
+			B int
+			C int
+		)
+
+		var (
+			timer           = time.NewTimer(10 * time.Second)
+			max       int32 = 3
+			done            = make(chan struct{})
+			running   int32 = 0
+			waitForUs       = func() error {
+				if atomic.AddInt32(&running, 1) == max {
```

Comment:
We typically use [go.uber.org/atomic](https://pkg.go.dev/go.uber.org/atomic) for atomics to prevent unintentional direct access to the atomic variable.

```
var running atomic.Int32

// ...

if running.Inc() == Max { ... }
```

Reply:
Roger, I was thinking about using that but didn't want to add another dependency unless you all were okay with it.